pkg/controller/signup_test.go

```diff
@@ -95,16 +94,16 @@ func (s *TestSignupSuite) TestSignupPostHandler() {
 		require.NoError(s.T(), err)
 		expectedUserID := ob.String()
 		ctx.Set(context.SubKey, expectedUserID)
-		ctx.Set(context.UsernameKey, "bill")
+		ctx.Set(context.UsernameKey, "bill@kubesaw")
 		ctx.Set(context.EmailKey, expectedUserID+"@test.com")
 		signup := testusersignup.NewUserSignup(
-			testusersignup.WithName("bill"),
-			IncompleteUserSignupCondition(),
+			testusersignup.WithEncodedName("bill@kubesaw"),
+			testusersignup.SignupIncomplete("test_reason", "test_message"),
 		)
 
 		svc.MockSignup = func(ctx *gin.Context) (*crtapi.UserSignup, error) {
 			assert.Equal(s.T(), expectedUserID, ctx.GetString(context.SubKey))
-			assert.Equal(s.T(), "bill", ctx.GetString(context.UsernameKey))
```

Comment:
What is the purpose of these asserts here? As far as I understand the flow, they're only checking that we set the appropriate values in the context a couple of lines above. Is that really needed?

What we're really should be testing there, IMHO, is that the signup service has been used. So what about instead of all those asserts on the context, having a "signupCalled" boolean that would be set to true in the `svc.MockSignup` and checked for being true after the call to handler?

Or maybe that's assumed by the correct status code of the response and we don't need any of that either?

Reply:
TBH, (to keep my sanity intact) I didn't try to question the tests themself, I only modified the related changes (or dropped tests that didn't make sense anymore after my changes). 
There are many things weird, wrong, missing, or redundant in the unit tests, I hope that we will get to fixing them later as soon as the whole code is simplified/refactored. 
Anyway, I tried to improve the test here: ec1eefa

Reply:
and as for you question:
> What we're really should be testing there

As I understand it, it verifies that the signup handler is called properly an returns corresponding response values